spec/services/types_generator_spec.rb

```diff
@@ -160,4 +162,80 @@
       end
     end
   end
+
+  describe 'cocina mapping' do
+    non_other_work_types = WorkType.all.reject { |work_type| work_type.id == 'other' }
+
+    let(:generator) { described_class.new(work: work) }
+    let(:types_to_genres) { generator.send(:types_to_genres) }
+    let(:types_to_resource_types) { generator.send(:types_to_resource_types) }
+
+    describe 'for work types' do
+      # NOTE: the Other type has no mappings
+      let(:work_type_labels) { non_other_work_types.map(&:label) }
+
+      it 'has one genre for each' do
+        # NOTE: General mappings do not correspond to a particular type
+        expect(work_type_labels).to match_array(types_to_genres.keys - ['General'])
+      end
+
+      it 'has one resource type for each' do
+        # NOTE: General mappings do not correspond to a particular type
+        expect(work_type_labels).to match_array(types_to_resource_types.keys - ['General'])
+      end
+    end
+
+    describe 'for subtypes' do
+      non_other_work_types.each do |work_type|
+        context "with type #{work_type.label}" do
+          let(:known_genreless) do
+            case work_type.label
+            when 'Data'
+              ['Documentation']
+            when 'Text'
+              ['Policy brief']
+            else
+              []
+            end
+          end
```

Comment:
Yep, lots of exceptions and special cases in our mappings.

Reply:
You mean metadata does not have consistent rules??

Reply:
FUNNY THAT